daemon/container_operations_unix.go

```diff
@@ -245,74 +278,104 @@ func (daemon *Daemon) setupSecretDir(c *container.Container) (setupErr error) {
 	return nil
 }
 
+func (daemon *Daemon) remountConfigDir(c *container.Container) error {
+	localMountPath := c.ConfigsDirPath()
+
+	label.Relabel(localMountPath, c.MountLabel, false)
+
+	rootIDs := daemon.idMappings.RootPair()
+	tmpfsOwnership := fmt.Sprintf("uid=%d,gid=%d", rootIDs.UID, rootIDs.GID)
+
+	// remount configs ro
+	if err := mount.Mount("tmpfs", localMountPath, "tmpfs", "remount,ro,"+tmpfsOwnership); err != nil {
```

Comment:
Just some thoughts while reading this;

- should the `tmpfs` have a _size_ specified? (based on the size of the configs/secrets)?
- (off topic); would it be possible in future to have non-readonly configs (probably tricky)?  - thinking of applications that use a configuration, but may add runtime configuration to the initial configuration.